### PR TITLE
get-authz-token-before-file-upload

### PR DESCRIPTION
**Pull request by Manoj-nathwani**
https://fjelltopp.atlassian.net/jira/software/projects/ADX/boards/4?selectedIssue=ADX-647

# Problem
- We were requesting the authz token on page load
- This meant if you open the file upload page, leave it open for 30mins, the token would expire when you finally got around to selecting a file to upload

# Solution
- We now request a new auth token at the time of actually uploading something

### CODE REVIEW

File: ckanext/unaids/react/components/FileInputComponent/src/FileUploader.js [hunk 1]
```diff
@@ -1,37 +1,55 @@
...
     setUploadProgress, setUploadFileName, setHiddenInputs,
     setUploadError
 }) {
 
+    const getAuthToken = () =>
+        axios.post(
+            '/api/3/action/authz_authorize',
+            { scopes: `obj:ckan/${datasetId}/*:write` },
+            { withCredentials: true }
+        ).then(res => res.data.result.token)
```
Comment (tomeksabala) [inline ckanext/unaids/react/components/FileInputComponent/src/FileUploader.js:17]: I think you're not handling `authz_authorize` error anywhere. Or is it going to be caught by:
``` 
            .catch(error => {
                setUploadError({
                    error: ckan.i18n._('Server Error'),
                    description: ckan.i18n._('An unknown server error has occurred.')
                });
                throw error;
            })
```
later down the line?

**Review by tomeksabala — CHANGES_REQUESTED** [review]

File: ckanext/unaids/react/components/FileInputComponent/src/FileUploader.js (hunk 1 above)
Comment (Manoj-nathwani) [reply ckanext/unaids/react/components/FileInputComponent/src/FileUploader.js:17]: Yep that's the case

Comment (tomeksabala) [reply ckanext/unaids/react/components/FileInputComponent/src/FileUploader.js:17]: @Manoj-nathwani Why not do it in the same way as in batch upload then?
image

**Review by tomeksabala — APPROVED** [review]
Ok!

**Pull request merged**